src/controllers/dlgprefcontrollers.cpp

```diff
@@ -256,6 +268,32 @@ void DlgPrefControllers::setupControllerWidgets() {
         temp.setBold(pController->isOpen());
         pControllerTreeItem->setFont(0, temp);
     }
+
+    // Update visibility based on checkbox state
+    updateControllerVisibility();
+}
+
+void DlgPrefControllers::updateControllerVisibility() {
+    bool hideUnenabled = m_pConfig->getValue(kHideUnenabledControllersCfgKey, false);
+
+    // Use the controller pages to match tree items with controllers
+    // since they are built together and always in sync
+    for (int i = 0; i < m_controllerTreeItems.size() && i < m_controllerPages.size(); ++i) {
+        QTreeWidgetItem* pTreeItem = m_controllerTreeItems.at(i);
+        DlgPrefController* pControllerDlg = m_controllerPages.at(i);
+
+        if (pTreeItem && pControllerDlg && pControllerDlg->controller()) {
```

Comment:
The `DlgPrefController` ctor does not assert a valid Controller* and there are no issues with that so I think we don't to check here either
```suggestion
        if (pTreeItem && pControllerDlg) {
```